From 12aa59d493c2db012e81712667057716539c54c4 Mon Sep 17 00:00:00 2001
From: icg_intg <icg_intg@intel.com>
Date: Wed, 2 May 2018 10:13:21 +0800
Subject: [PATCH 1885/2328] isys & psys ipu4p package lib2600b0 for commit-id
 1593412

FW_Manifest_ID: 3b35f83db390434aa7cdb34e7bd3220cdabac0de

FW_Build_Branch: IPU4_master

FW_Build_Topic: auto_merge_queue

library version for ipu4p: commit-id 1593412

FW Bundle patch number of this release :  325239

FW tag : scci_IPU4_master_20180427_2117

Change-Id: If2a36b561b76c5b4b36900620463a7881992f902
Depends-on: Id845d547f420bd4b0cb9407ce45b6d3cd6587b93
Depends-on: Icecf8ac8a77a37050801f4691322a61f8f45c23c
Depends-on: I427afb682cd957c60b798db4c5b398a7c906e071
Signed-off-by: icg_intg <icg_intg@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 .../intel/ipu4/ipu4p-css/ia_css_fw_pkg_release.h   |   2 +-
 .../lib2600/cpd_binary/ia_css_fw_pkg_release.h     |   2 +-
 .../interface/ia_css_isys_fw_bridged_types.h       |   1 +
 .../interface/ia_css_isysapi_proxy_region_defs.h   |   8 --
 .../isysapi/interface/ia_css_isysapi_types.h       |   1 +
 .../lib2600/reg_dump/src/reg_dump_generic_bridge.c |   7 --
 .../lib/CNL_program_group/ia_css_fw_pkg_release.h  |   2 +-
 .../lib/ICL_program_group/ia_css_fw_pkg_release.h  |   2 +-
 .../interface/ipu_device_gp_properties_types.h     | 103 ---------------------
 .../lib/reg_dump/src/reg_dump_generic_bridge.c     |   7 --
 .../lib/vied_parameters/vied_parameters.mk         |   1 -
 11 files changed, 6 insertions(+), 130 deletions(-)

diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/ia_css_fw_pkg_release.h b/drivers/media/pci/intel/ipu4/ipu4p-css/ia_css_fw_pkg_release.h
index 795c783..29eef75 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/ia_css_fw_pkg_release.h
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/ia_css_fw_pkg_release.h
@@ -11,4 +11,4 @@
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 */
-#define IA_CSS_FW_PKG_RELEASE  0x20180420
+#define IA_CSS_FW_PKG_RELEASE  0x20180427
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/cpd_binary/ia_css_fw_pkg_release.h b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/cpd_binary/ia_css_fw_pkg_release.h
index 795c783..29eef75 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/cpd_binary/ia_css_fw_pkg_release.h
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/cpd_binary/ia_css_fw_pkg_release.h
@@ -11,4 +11,4 @@
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 */
-#define IA_CSS_FW_PKG_RELEASE  0x20180420
+#define IA_CSS_FW_PKG_RELEASE  0x20180427
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isys_fw_bridged_types.h b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isys_fw_bridged_types.h
index aff4cfb9..da62337 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isys_fw_bridged_types.h
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isys_fw_bridged_types.h
@@ -61,6 +61,7 @@ struct ia_css_isys_resolution_comm {
  * @out_buf_id: Points to output pin buffer - buffer identifier
  * @addr: Points to output pin buffer - CSS Virtual Address
  * @compress: Request frame compression (1), or  not (0)
+ * This must be the same as ia_css_isys_output_pin_info_comm::reserve_compression
  */
 struct ia_css_isys_output_pin_payload_comm {
 	aligned_uint64(ia_css_return_token, out_buf_id);
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isysapi_proxy_region_defs.h b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isysapi_proxy_region_defs.h
index 1d1dbf3..0fb8310 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isysapi_proxy_region_defs.h
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isysapi_proxy_region_defs.h
@@ -110,13 +110,5 @@ struct ia_css_proxy_write_region_description ipu4p_b0_reg_write_desc[N_IPU4P_B0_
 
 #endif /*defined(IPU4P_B0_PROXY_INT)*/
 
-/*
- */
-
-
-
-/*
- */
-
 
 #endif /* __IA_CSS_ISYSAPI_PROXY_REGION_DEFS_H */
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isysapi_types.h b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isysapi_types.h
index cc00cb6..4a3903f 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isysapi_types.h
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/isysapi/interface/ia_css_isysapi_types.h
@@ -89,6 +89,7 @@ struct ia_css_isys_resolution {
  * @out_buf_id: Points to output pin buffer - buffer identifier
  * @addr: Points to output pin buffer - CSS Virtual Address
  * @compressed: Request frame compression (1), or  not (0)
+ * This must be the same as ia_css_isys_output_pin_info::reserve_compression
  */
 struct ia_css_isys_output_pin_payload {
 	ia_css_return_token out_buf_id;
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/reg_dump/src/reg_dump_generic_bridge.c b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/reg_dump/src/reg_dump_generic_bridge.c
index dab9c66..18d0c98 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/reg_dump/src/reg_dump_generic_bridge.c
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600/reg_dump/src/reg_dump_generic_bridge.c
@@ -26,15 +26,8 @@
 #define REG_DUMP_TRACE_METHOD IA_CSS_TRACE_METHOD_NATIVE
 #define REG_DUMP_TRACE_LEVEL_VERBOSE IA_CSS_TRACE_LEVEL_ENABLED
 
-#ifdef USE_SSID_BUTTRESS
- * seen from the host; these addresses already contain the ISYS or PSYS offset.
- */
-#define REG_DUMP_READ_REGISTER(addr)\
-	 vied_subsystem_load_32(IPU_DEVICE_BUTTRESS, addr)
-#else
 /* SSID value is defined in test makefiles as either isys0 or psys0 */
 #define REG_DUMP_READ_REGISTER(addr)    vied_subsystem_load_32(SSID, addr)
-#endif
 
 #define REG_DUMP_PRINT_0(...) \
 EXPAND_VA_ARGS(IA_CSS_TRACE_0(REG_DUMP, VERBOSE, __VA_ARGS__))
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/CNL_program_group/ia_css_fw_pkg_release.h b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/CNL_program_group/ia_css_fw_pkg_release.h
index 795c783..29eef75 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/CNL_program_group/ia_css_fw_pkg_release.h
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/CNL_program_group/ia_css_fw_pkg_release.h
@@ -11,4 +11,4 @@
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 */
-#define IA_CSS_FW_PKG_RELEASE  0x20180420
+#define IA_CSS_FW_PKG_RELEASE  0x20180427
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/ICL_program_group/ia_css_fw_pkg_release.h b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/ICL_program_group/ia_css_fw_pkg_release.h
index 795c783..29eef75 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/ICL_program_group/ia_css_fw_pkg_release.h
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/ICL_program_group/ia_css_fw_pkg_release.h
@@ -11,4 +11,4 @@
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 */
-#define IA_CSS_FW_PKG_RELEASE  0x20180420
+#define IA_CSS_FW_PKG_RELEASE  0x20180427
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/devices/interface/ipu_device_gp_properties_types.h b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/devices/interface/ipu_device_gp_properties_types.h
index b57a2fc..acd9111 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/devices/interface/ipu_device_gp_properties_types.h
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/devices/interface/ipu_device_gp_properties_types.h
@@ -71,73 +71,6 @@ enum ipu_device_gp_isa_value {
 	IPU_DEVICE_GP_ISA_PAF_STR_PORT0 = 0,
 	IPU_DEVICE_GP_ISA_PAF_STR_PORT1 = 1,
 
-	/* scaler port block options */
-	IPU_DEVICE_GP_ISA_SCALER_PORT_UNBLOCK = 0,
-	IPU_DEVICE_GP_ISA_SCALER_PORT_BLOCK = 1,
-
-	IPU_DEVICE_GP_ISA_STATIC_SCALED_OUT_DEMUX_SEL_S2V = 0,
-	IPU_DEVICE_GP_ISA_STATIC_SCALED_OUT_DEMUX_SEL_PSA = 1,
-
-	/* Muxes/demuxes */
-	/* 0 - to ISL.S2V; 1 - to PSA */
-	IPU_DEVICE_GP_ISA_STATIC_ISA_ORIG_OUT_DEMUX_SEL_S2V = 0,
-	IPU_DEVICE_GP_ISA_STATIC_ISA_ORIG_OUT_DEMUX_SEL_PSA = 1,
-	/* 0 - to ISL.S2V; 1 - to PSA */
-	IPU_DEVICE_GP_ISA_STATIC_ISA_SCALED_A_OUT_DEMUX_SEL_S2V = 0,
-	IPU_DEVICE_GP_ISA_STATIC_ISA_SCALED_A_OUT_DEMUX_SEL_PSA = 1,
-	/* 0 - Input Correction input. 1 - B2B mux input */
-	IPU_DEVICE_GP_ISA_STATIC_AWB_MUX_SEL_INPUT_CORR = 0,
-	IPU_DEVICE_GP_ISA_STATIC_AWB_MUX_SEL_B2B = 1,
-	/* 0 - to Lsc , 1 - to Dpc , 2 - to X2b */
-	IPU_DEVICE_GP_ISA_STATIC_ISA_INPUT_CORR_DEMUX_SEL_LSC = 0,
-	IPU_DEVICE_GP_ISA_STATIC_ISA_INPUT_CORR_DEMUX_SEL_DPC = 1,
-	IPU_DEVICE_GP_ISA_STATIC_ISA_INPUT_CORR_DEMUX_SEL_X2B = 2,
-	/* 0 - Input correction, 1 - Dpc , 2 - X2b */
-	IPU_DEVICE_GP_ISA_STATIC_LSC_MUX_SEL_INPUT_CORR = 0,
-	IPU_DEVICE_GP_ISA_STATIC_LSC_MUX_SEL_DPC = 1,
-	IPU_DEVICE_GP_ISA_STATIC_LSC_MUX_SEL_X2B = 2,
-	/* 0 - to B2b, 1 - to Dpc , 2 - X2b*/
-	IPU_DEVICE_GP_ISA_STATIC_LSC_DEMUX_SEL_B2B = 0,
-	IPU_DEVICE_GP_ISA_STATIC_LSC_DEMUX_SEL_DPC = 1,
-	IPU_DEVICE_GP_ISA_STATIC_LSC_DEMUX_SEL_X2B = 2,
-	/* 0 - Lsc, 1 - Input correction , 2 - X2b */
-	IPU_DEVICE_GP_ISA_STATIC_DPC_MUX_SEL_LSC = 0,
-	IPU_DEVICE_GP_ISA_STATIC_DPC_MUX_SEL_INPUT_CORR = 1,
-	IPU_DEVICE_GP_ISA_STATIC_DPC_MUX_SEL_X2B = 2,
-	/* 0 - to Lsc, 1 - to B2b , 2 - to X2b */
-	IPU_DEVICE_GP_ISA_STATIC_DPC_DEMUX_SEL_LSC = 0,
-	IPU_DEVICE_GP_ISA_STATIC_DPC_DEMUX_SEL_B2B = 1,
-	IPU_DEVICE_GP_ISA_STATIC_DPC_DEMUX_SEL_X2B = 2,
-	/* 0 - Lsc, 1 - X2b, 2 - Input correction */
-	IPU_DEVICE_GP_ISA_STATIC_X2B_MUX_SEL_LSC = 0,
-	IPU_DEVICE_GP_ISA_STATIC_X2B_MUX_SEL_X2B = 1,
-	IPU_DEVICE_GP_ISA_STATIC_X2B_MUX_SEL_INPUT_CORR = 2,
-	/* 0 - Through X2B S2V RGBIR, 1 - Bypass */
-	IPU_DEVICE_GP_ISA_STATIC_SVE_RGBIR_BP_MUX_DEMUX_SEL_SVE_RGBIR = 0,
-	IPU_DEVICE_GP_ISA_STATIC_SVE_RGBIR_BP_MUX_DEMUX_SEL_BYPASS = 1,
-	/* 0 - X2B SVE RGBIR, 1- X2B MD */
-	IPU_DEVICE_GP_ISA_STATIC_IR_DEPTH_MUX_SEL_SVE_RGBIR = 0,
-	IPU_DEVICE_GP_ISA_STATIC_IR_DEPTH_MUX_SEL_MD = 1,
-	/* 0 - to Lsc, 1 - to Dpc, 2 - to B2b */
-	IPU_DEVICE_GP_ISA_STATIC_X2B_DEMUX_SEL_LSC = 0,
-	IPU_DEVICE_GP_ISA_STATIC_X2B_DEMUX_SEL_DPC = 1,
-	IPU_DEVICE_GP_ISA_STATIC_X2B_DEMUX_SEL_B2B = 2,
-	/* 0 - Lsc, 1 - Dpc , 2 - X2b */
-	IPU_DEVICE_GP_ISA_STATIC_B2B_MUX_SEL_LSC = 0,
-	IPU_DEVICE_GP_ISA_STATIC_B2B_MUX_SEL_DPC = 1,
-	IPU_DEVICE_GP_ISA_STATIC_B2B_MUX_SEL_X2B = 2,
-	/* 0 - External PAF CH0/1; 1 - DPC extracted PAF CH0/1; 2 - X2B extracted PAF CH0/ Black Box CH1 */
-	IPU_DEVICE_GP_ISA_STATIC_PAF_SRC_SEL_EXT_PAF = 0,
-	IPU_DEVICE_GP_ISA_STATIC_PAF_SRC_SEL_DPC = 1,
-	IPU_DEVICE_GP_ISA_STATIC_PAF_SRC_SEL_X2B = 2,
-	/* 0 - from R2I; 1 - from B2B - TODO remove after SDK 0.5 as it was removed from design */
-	IPU_DEVICE_GP_ISA_STATIC_ISA_ORIG_OUT_MUX_SEL_R2I = 0,
-	IPU_DEVICE_GP_ISA_STATIC_ISA_ORIG_OUT_MUX_SEL_B2B = 1,
-	/* Blockers */
-	IPU_DEVICE_GP_ISA_STATIC_PORT_BLK_UNBLOCK = 0,
-	IPU_DEVICE_GP_ISA_STATIC_PORT_BLK_BLOCK = 1,
-
-
 	/* sis port block options */
 	IPU_DEVICE_GP_ISA_SIS_PORT_UNBLOCK = 0,
 	IPU_DEVICE_GP_ISA_SIS_PORT_BLOCK = 1,
@@ -158,42 +91,6 @@ enum ipu_device_gp_psa_value {
 	/* PSA_V2S_RGB_4_DEMUX */
 	IPU_DEVICE_GP_PSA_DEMUX_POST_V2S_RGB_4_TO_GTM = 0,
 	IPU_DEVICE_GP_PSA_DEMUX_POST_V2S_RGB_4_TO_ACM = 1,
-	 * fixed function MAS. Choose between pixel stream and
-	 * delta stream as BNLM output (gpreg 1)
-	 */
-	IPU_DEVICE_GP_PSA_1_NOISE_MUX_BNLM_PIXELS = 0,
-	IPU_DEVICE_GP_PSA_1_NOISE_MUX_DELTA_STREAM = 1,
-	/* enable/disable BNLM Pixel Block (gpreg 2) */
-	IPU_DEVICE_GP_PSA_1_BNLM_PIXEL_STREAM_BLOCK_DISABLE = 0,
-	IPU_DEVICE_GP_PSA_1_BNLM_PIXEL_STREAM_BLOCK_ENABLE = 1,
-	/* enable/disable BNLM delta stream (gpreg 3) */
-	IPU_DEVICE_GP_PSA_1_BNLM_DELTA_STREAM_BLOCK_DISABLE = 0,
-	IPU_DEVICE_GP_PSA_1_BNLM_DELTA_STREAM_BLOCK_ENABLE = 1,
-	/* choose BNLM output to XNR or to WB/DM (gpreg 0) */
-	IPU_DEVICE_GP_PSA_2_BNLM_TO_XNR = 0,
-	IPU_DEVICE_GP_PSA_2_BNLM_TO_WB_DM = 1,
-	/* choose direction of output from vec2str 4 (gpreg 4) */
-	IPU_DEVICE_GP_PSA_2_V2S_RGB_4_TO_GTC = 0,
-	IPU_DEVICE_GP_PSA_2_V2S_RGB_4_TO_ACM = 1,
-	IPU_DEVICE_GP_PSA_2_V2S_RGB_4_TO_VCSC = 2,
-	IPU_DEVICE_GP_PSA_2_V2S_RGB_4_TO_GSTAR = 3,
-	/* enable/disable VCSC input block (gpreg 7) */
-	IPU_DEVICE_GP_PSA_2_VCSC_INPUT_BLOCK_DISABLE = 0,
-	IPU_DEVICE_GP_PSA_2_VCSC_INPUT_BLOCK_ENABLE = 1,
-	/* enable/disable XNR5 bypass block (gpreg 8) */
-	IPU_DEVICE_GP_PSA_2_XNR5_BP_BLOCK_DISABLE = 0,
-	IPU_DEVICE_GP_PSA_2_XNR5_BP_BLOCK_ENABLE = 1,
-	/* choose to use VCSC or bypass it (gpreg 5) */
-	IPU_DEVICE_GP_PSA_3_MUX_USE_VCSC = 0,
-	IPU_DEVICE_GP_PSA_3_MUX_BP_VCSC = 1,
-	/* choose to use XNR5 or bypass it (gpreg 6) */
-	IPU_DEVICE_GP_PSA_3_MUX_USE_XNR5 = 0,
-	IPU_DEVICE_GP_PSA_3_MUX_BP_XNR5 = 1,
-	/* choose which input to use for the BNLM acc */
-	IPU_DEVICE_GP_PSA_1_BNLM_IN_MUX_V2S = 0,
-	IPU_DEVICE_GP_PSA_1_BNLM_IN_MUX_ISA_DOWNSCALED = 1,
-	IPU_DEVICE_GP_PSA_1_BNLM_IN_MUX_ISA_UNSCALED = 2,
-	IPU_DEVICE_GP_PSA_CONF_INVALID = 0xFF
 };
 
 enum ipu_device_gp_isl_value {
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/reg_dump/src/reg_dump_generic_bridge.c b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/reg_dump/src/reg_dump_generic_bridge.c
index dab9c66..18d0c98 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/reg_dump/src/reg_dump_generic_bridge.c
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/reg_dump/src/reg_dump_generic_bridge.c
@@ -26,15 +26,8 @@
 #define REG_DUMP_TRACE_METHOD IA_CSS_TRACE_METHOD_NATIVE
 #define REG_DUMP_TRACE_LEVEL_VERBOSE IA_CSS_TRACE_LEVEL_ENABLED
 
-#ifdef USE_SSID_BUTTRESS
- * seen from the host; these addresses already contain the ISYS or PSYS offset.
- */
-#define REG_DUMP_READ_REGISTER(addr)\
-	 vied_subsystem_load_32(IPU_DEVICE_BUTTRESS, addr)
-#else
 /* SSID value is defined in test makefiles as either isys0 or psys0 */
 #define REG_DUMP_READ_REGISTER(addr)    vied_subsystem_load_32(SSID, addr)
-#endif
 
 #define REG_DUMP_PRINT_0(...) \
 EXPAND_VA_ARGS(IA_CSS_TRACE_0(REG_DUMP, VERBOSE, __VA_ARGS__))
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/vied_parameters/vied_parameters.mk b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/vied_parameters/vied_parameters.mk
index 5e89033..834a1a4 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/vied_parameters/vied_parameters.mk
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/lib2600psys/lib/vied_parameters/vied_parameters.mk
@@ -72,6 +72,5 @@ VIED_PARAMETERS_FW_CPPFLAGS +=  $(VIED_PARAMETERS_SUPPORT_CPPFLAGS)
 #For IPU interface
 include $(MODULES_DIR)/fw_abi_common_types/cpu/fw_abi_cpu_types.mk
 VIED_PARAMETERS_HOST_CPPFLAGS += $(FW_ABI_COMMON_TYPES_HOST_CPPFLAGS)
-endif
 
 VIED_PARAMETERS_FW_CPPFLAGS   += $(FW_ABI_COMMON_TYPES_FW_CPPFLAGS)
-- 
2.7.4

